cryptol-remote-api/Dockerfile

```diff
@@ -27,7 +27,11 @@ RUN if ${PORTABILITY}; then \
         ghcup install ghc ${GHCVER_BOOTSTRAP} && \
         ghcup set ghc ${GHCVER_BOOTSTRAP} && \
         cabal v2-update && \
-        cabal v2-install alex happy-1.20.1.1 && \
+        # We pin specific versions of alex and happy that are necessary to build
+        # GHC. In general, we don't want to build the latest versions of each
+        # tool, as GHC may not support them (see, for example,
+        # https://gitlab.haskell.org/ghc/ghc/-/issues/24302).
+        cabal v2-install alex-3.4.0.1 happy-1.20.1.1 && \
```

Comment:
Should we add a comment up near the GHCVER definition indicating that these versions will likely need to be updated if the GHC version changes?  And also perhaps a comment here referencing issue #1599?  Mostly in service of helping inform future maintainers.

Reply:
Yes, good idea. I've also factored out these version numbers into dedicated `ALEXVER`/`HAPPYVER` arguments so that we can put them alongside the definition of `GHCVER_BOOTSTRAP`, which is closely related.